paasta_tools/cleanup_kubernetes_jobs.py

```diff
@@ -0,0 +1,188 @@
+#!/usr/bin/env python
+# Copyright 2015-2016 Yelp Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Usage: ./cleanup_marathon_jobs.py [options]
+
+Clean up marathon apps that aren't supposed to run on this cluster by deleting them.
+
+Gets the current app list from marathon, and then a 'valid_app_list'
+via utils.get_services_for_cluster
+
+If an app in the marathon app list isn't in the valid_app_list, it's
+deleted.
+
+Command line options:
+
+- -d <SOA_DIR>, --soa-dir <SOA_DIR>: Specify a SOA config dir to read from
+- -v, --verbose: Verbose output
+- -t <KILL_THRESHOLD>, --kill-threshold: The decimal fraction of apps we think
+    is sane to kill when this job runs
+- -f, --force: Force the killing of apps if we breach the threshold
+"""
+import argparse
+import logging
+import sys
+import traceback
+from contextlib import contextmanager
+
+from kubernetes.client import V1Deployment
+from kubernetes.client import V1StatefulSet
+
+from paasta_tools.kubernetes.tools import Application
+from paasta_tools.kubernetes.tools import list_namespaced_applications
+from paasta_tools.kubernetes_tools import KubeClient
+from paasta_tools.utils import _log
+from paasta_tools.utils import DEFAULT_SOA_DIR
+from paasta_tools.utils import get_services_for_cluster
+from paasta_tools.utils import load_system_paasta_config
+# import pysensu_yelp
+# from paasta_tools.monitoring_tools import send_event
+# import pysensu_yelp
+
+
+log = logging.getLogger(__name__)
+application_types = [V1StatefulSet, V1Deployment]
```

Comment:
https://www.python.org/dev/peps/pep-0008/#constants
> Constants are usually defined on a module level and written in all capital letters with underscores separating words. 
 Examples include MAX_OVERFLOW and TOTAL.